### PR TITLE
change index in tests as years are cast to datetimes

### PR DESCRIPTION
**Pull request by lewisjared**
@znicholls This should resolve the failing tests

### CODE REVIEW

**Comment by znicholls** [comment]
@rgieseke I've pushed this one through as it's tiny, hope that's all good

**Comment by lewisjared** [comment]
@znicholls it should pass this time. Obviously, ScmTimeSeries can't guarantee to preserve meta row ordering.

File: tests/test_io.py [hunk 1]
```diff
@@ -2614,6 +2614,10 @@ def test_magicc_data_append(mock_read_and_return_metadata_df):
     expected_metadata = deepcopy(tmetadata_init)
     expected_metadata.update(tmetadata_append)
     assert mdata.metadata == expected_metadata
+
+    # update the tdf_init index as year values are converted to datetimes
+    tdf_init.index = mdata.timeseries().columns
+    tdf_append.index = mdata.timeseries().columns
     pd.testing.assert_frame_equal(
         mdata.timeseries().reset_index(drop=True),
```
Comment (znicholls) [inline tests/test_io.py]:
```suggestion
        mdata.timeseries().sort_index().reset_index(),
```

File: tests/test_io.py [hunk 2]
```diff
@@ -2614,6 +2614,10 @@ def test_magicc_data_append(mock_read_and_return_metadata_df):
...
         tdf_init.T.append(tdf_append.T).reset_index(drop=True),
```
Comment (znicholls) [inline tests/test_io.py]:
```suggestion
        tdf_init.T.append(tdf_append.T).sort_index().reset_index(),
```

Comment (lewisjared) [reply tests/test_io.py]: tdf_init.T has a junk index so sorting didn't work. Just needed to explicitly sort mdata.timeseries() as that the one which was inconcistent

Comment (lewisjared) [reply tests/test_io.py]: Also needed the drop since the indexes aren't the same

**Comment by lewisjared** [comment]
@znicholls about time

**Comment by znicholls** [comment]
hang on I think we can make this a bit more obvious

**Comment by znicholls** [comment]
I'll make a PR now

**Comment by lewisjared** [comment]
@znicholls Good to go?

**Pull request merged**